src/extension.ts

```diff
@@ -1,44 +1,37 @@
-// /********************************************************
-// *                                                       *
-// *   Copyright (C) Microsoft. All rights reserved.       *
-// *                                                       *
-// ********************************************************/
-import * as sarif from "sarif";
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
 import { commands, ExtensionContext } from "vscode";
 import { CodeFlowCodeLensProvider } from "./CodeFlowCodeLens";
 import { CodeFlowDecorations } from "./CodeFlowDecorations";
 import { ExplorerController } from "./ExplorerController";
 import { FileConverter } from "./FileConverter";
-import { FileMapper } from "./FileMapper";
 import { LogReader } from "./LogReader";
 import { SVCodeActionProvider } from "./SVCodeActionProvider";
 import { Utilities } from "./Utilities";
+import { ResultsListController } from "./ResultsListController";
 
 /**
  * This method is called when the extension is activated.
  * Creates the explorer, reader, provider
  * Process any open SARIF Files
  */
-export function activate(context: ExtensionContext) {
-    // Create the launch Explorer command
-    context.subscriptions.push(
-        commands.registerCommand(ExplorerController.ExplorerLaunchCommand, () => {
-            ExplorerController.Instance.createWebview();
-        }));
+export async function activate(context: ExtensionContext): Promise<void> {
+    Utilities.initialize(context);
+    FileConverter.initialize(context);
 
-    context.subscriptions.push(
-        commands.registerCommand(ExplorerController.SendCFSelectionToExplorerCommand, (id: string) => {
-            CodeFlowDecorations.updateCodeFlowSelection(id);
-            ExplorerController.Instance.setSelectedCodeFlow(id);
-        }));
+    const explorerController: ExplorerController = new ExplorerController(context);
+    context.subscriptions.push(explorerController);
```

Comment:
We could do ```
context.subscriptions.push(
    explorerController,
    codeActionProvider,
    ....and so on...
)
```

Reply:
We discussed why this isn't a great idea over the Teams\Skype\Zoom

---
In reply to: [397540093](https://github.com/microsoft/sarif-vscode-extension/pull/211#discussion_r397540093) [](ancestors = 397540093)